pkg/edgeview/src/system.go

```diff
@@ -1306,6 +1315,140 @@ func runTechSupport(cmds cmdOpt, isLocal bool) {
 	}
 }
 
+// edgeview request debug container to generate collect-info and
+// later to remove the tar.gz file
+// the protocol is for edgeview to create a file
+// /run/edgeview/edgeview-request-collect-info, and debug container
+// run 'collect-info.sh' to generate the tar.gz file, when the job is
+// done, remove the file /run/edgeview/edgeview-request-collect-info.
+// edgeview can also request to remove the tar.gz file by creating
+// /run/edgeview/edgeview-request-remove-tar.gz file, and debug container
+// will remove the generated tar.gz file.
+func runCollectInfo() {
+	if _, err := os.Stat(requestCollectInfoFile); err == nil {
+		fmt.Printf("runCollectInfo, collect info request already in progress\n")
+		return
+	}
+
+	startTime := time.Now() // Record the start time
+
+	err := touchRequestFile(requestCollectInfoFile, "")
+	if err != nil {
+		fmt.Printf("runCollectInfo, can not create collect info request file\n")
+		return
+	}
+
+	err = monitorFileDeletion(requestCollectInfoFile, startTime, 20*time.Minute)
+	if err != nil {
+		log.Errorf("runCollectInfo: monitor file deletion error %v", err)
+		fmt.Println("runCollectInfo, monitor file deletion error")
+		return
+	}
+
+	targzFilename, err := findEveInfoFile(startTime)
+	if err != nil {
+		log.Errorf("runCollectInfo: can not find the eve info file, %v", err)
+		fmt.Printf("runCollectInfo, can not find the eve info file\n")
+		return
+	}
+
+	// Copy the info tar.gz file to the client side
+	err = runCopy("cp/"+targzFilename, nil)
+	if err != nil {
+		log.Errorf("runCollectInfo: copy kube file error %v", err)
+	}
+
+	// ask to remove this tar.gz file
+	err = touchRequestFile(requestRemoveInfoGZFile, targzFilename)
+	if err != nil {
+		log.Errorf("runCollectInfo: generate remove request file, %v", err)
+	}
+}
+
+func monitorFileDeletion(filePath string, startTime time.Time, timeout time.Duration) error {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Errorf("monitorFileDeletion: %v", err)
+		return err
+	}
+	defer watcher.Close()
+
+	done := make(chan bool)
```

Comment:
How about changing channel value to error and then returning the error from this function?
This would preserve errors like "Timeout exceeded", which could be printed to the console instead of potentially confusing message "can not find the eve info file".

Reply:
sure.